File: src/main/java/application/history/history/presentation/interfaces/api/GetAllHistoriesController.java
package application.history.history.presentation.interfaces.api;

import application.history.history.application.services.GetAllHistoriesService;
import application.history.history.domain.entities.History;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class GetAllHistoriesController {
    private final GetAllHistoriesService service;

    public GetAllHistoriesController(GetAllHistoriesService getAllHistory) {
        this.service = getAllHistory;
    }

    @GetMapping("/histories")
    public ResponseEntity<Iterable<History>> getAllHistories() {
        return ResponseEntity.ok(service.run());
    }
}

File: src/main/kotlin/com/example/history/presentation/interfaces/api/GetAllHistoriesController.kt
package com.example.history.presentation.interfaces.api

import com.example.history.application.services.FindHistoryUseCase
import com.example.history.domain.entities.History
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/histories")
class GetAllHistoriesController(private val service: FindHistoryUseCase) {

    @GetMapping
    fun getHistories(): Iterable<History> {
        return service.findAll()
    }
}
